core/dbt/parser/models.py

```diff
@@ -45,7 +45,7 @@ def __init__(self):
     def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
         if node.name == "model":
             self.num_model_def += 1
-            if not node.args.args[0].arg == "dbt":
+            if node.args.args and not node.args.args[0].arg == "dbt":
                 self.dbt_errors.append("'dbt' not provided for model as the first argument")
             if len(node.args.args) != 2:
```

Comment:
Maybe something like this too?
```suggestion
            if not node.args.args or len(node.args.args) != 2:
```

Reply:
In this case it is not necessary because `node.args.args` is an empty list, so `len` on it works. Checking `node.args.args and`  above is counting with the fact that an empty list is falsy.